cortex/secondary/call_duration.py

```diff
@@ -27,15 +31,38 @@ def call_duration(incoming=True, **kwargs):
             timestamp (int): The beginning of the window (same as kwargs['start']).
             value (float): The time spent in a call.
     """
-    incoming_dict = {True: 1, False: 2}
-    label = incoming_dict[incoming]
-    log.info('Loading raw calls data...')
-    _calls = calls(**kwargs)['data']
-    log.info('Computing call duration...')
-    _call_duration = np.sum([call['call_duration'] for call in _calls
-                             if call['call_type'] == label])
-    # if you have no call duration, this means you have no call data
-    # over the period, should return None
-    if _call_duration == 0:
-        _call_duration = None
-    return {'timestamp': kwargs['start'], 'value': _call_duration}
+    _calls = telephony(**kwargs)['data']
+
+    incoming = kwargs.get('incoming')
+
+    # optional incoming variable sets call_direction in accordance with
+    # previous version of call_duration, in which function returned
+    # incoming calls if incoming is True, else returned outgoing calls.
+
+    if incoming is not None:
+        if incoming is True:
+            call_direction = "incoming"
+        else:
+            call_direction = "outgoing"
+
+    if call_direction == "all":
+        call_duration = np.sum(call['duration'] for call in _calls)
+
+    elif call_direction == "incoming":
+        call_duration = np.sum(call['duration'] for call in _calls 
+                               if call['type'] == "incoming")
+    elif call_direction == "outgoing":
+        call_duration = np.sum(call['duration'] for call in _calls
+                               if call['type'] == "outgoing")
+    else:
+        call_duration = None
+        log.info('"' + call_direction + "' was passed but is not an acceptable argument. Acceptable arguments include 'all','incoming', or 'outgoing'")
+
+    # if you have no call duration of any kind, this means you have no call data
+    # in this case, return None.
+
+    if np.sum(call['duration'] for call in _calls) == 0:
```

Comment:
None should be returned if there is no call data (it is fine if there is call duration but the duration for that direction is 0).